src/packagedcode/rubygems.py

```diff
@@ -648,92 +650,127 @@ def normalize(gem_data, known_fields=known_fields):
+            # >>> name = s.split('= ')
+            # >>> name
+            # ['spec.name ', '"abc"']
+            # >>> name = name[1].strip('\'"%q{}')
+            # >>> name
+            # abc
+            name = individual.split('= ')
+            name = name[1].strip('\'"%q{}')
+            gemspec_data['name'] = name
+
+        # update the value of author
+        if 'authors' in individual:
+            # >>> s = 's.authors = ["abc", "pqr", "xyz"]'
+            # >>> result = s.split('= ')
+            # ['s.authors ', '["abc", "pqr", "xyz"]']
+            # >>> result[1].strip('[]')
+            # '"abc", "pqr", "xyz"'
+            authors = individual.split('= ')
+            authors = authors[1].strip('[]%q{}').replace('"', '')
+            gemspec_data['authors'] = authors
+
+        # update the value of email
+        if 'email' in individual:
+            email = individual.split('= ')
+            email = email[1].strip('[]%q{}').replace('"', '')
+            gemspec_data['email'] = email
+
+        # update the value of summary
+        if 'summary' in individual:
+            summary = individual.split('= ')
+            summary = summary[1].strip('\'"%q{}')
+            gemspec_data['summary'] = summary
+
+        # update the value of description
+        if 'description' in individual:
+            description = individual.split('= ')
+            description = description[1].strip('\'"%q{}')
+            gemspec_data['description'] = description
+
+        # update the value of homepage
+        if 'homepage' in individual and 'homepage_uri' not in individual:
+            homepage = individual.split('= ')
+            homepage = homepage[1].strip('\'"%q{}')
+            gemspec_data['homepage'] = homepage
+
+        # update the value of license
+        if 'license' in individual:
+            license = individual.split('= ')
+            license = license[1].strip('\'"%q{}')
+            gemspec_data['license'] = license
+
+        # update the value of dependencies
+        if 'dependency' in  individual:
+            # >>> s = 's.add_runtime_dependency(%q<mechanize>, [">= 0"])'
+            # >>> d = re.split("'|<|>", s)
+            # >>> d
+            # ['s.add_runtime_dependency(%q', 'mechanize', ', ["', '= 0"])']
+            dependency = re.split("'|\"|<|>", individual)
+            dependencies.append(dependency[1])
+
+    # stores dependencies list into dictionary
+    gemspec_data['dependencies'] = dependencies
+
+    return gemspec_data
+
+
+def build_packages_from_gemspec(gemspec_data):
     """
-    Represent a Gem specification.
+    Yield RubyGem Packages from gemspec.
     """
+    name = gemspec_data.get('name')
+    description = gemspec_data.get('description')
+    homepage_url = gemspec_data.get('homepage')
+    package = RubyGem(
+        name=name,
+        description=description,
+        homepage_url=homepage_url
+    )
+    package.dependencies = gemspec_data.get('dependencies')
```

Comment:
You will need to do a little more here to handle the dependencies here. See: https://github.com/nexB/scancode-toolkit/blob/72bef69a3c485380be164a58432dd8218ce4df84/src/packagedcode/models.py#L458

You have to create DependentPackage objects (similar to how you handle parties)

Reply:
I will look at this too :)